### DISCUSSION TITLE
Retry until failure

### DISCUSSION THREAD

**Discussion by barakadax** [nunit/nunit#4329] [Q&A]
Hi, so the retry attribute is great but I need to debug something that randomly fails, can be even once a week or so.
Am I allowed to add attribute to retry until failure?

**Comment by OsirisTerje** [comment]
There is no such attribute, but what prevents you from creating a test that loops until your system under test fails?
![image](https://user-images.githubusercontent.com/203432/229819650-6cd492be-1390-4f4f-b113-fffdf528724e.png)

**Reply by barakadax** [reply]
I found out it doesn't exists, I want to add something like that, is it fine?
I can add loop to the code and hope for the best but I want to have minimal changes in my code for debugging.

**Reply by barakadax** [reply]
Ok, so I wrote the code for the attribute I want to add but I get: `fatal: unable to access 'https://github.com/nunit/nunit.git/': The requested URL returned error: 403`
So I can't show the code :(

**Comment by barakadax** [comment]
https://github.com/nunit/nunit/pull/4330 can someone please help me with this?

**Reply by OsirisTerje** [reply]
@stevenaw @manfred-brands Comments please :-)

**Comment by barakadax** [comment]
Made https://github.com/nunit/nunit/pull/4330 a bit safer to use 😄
